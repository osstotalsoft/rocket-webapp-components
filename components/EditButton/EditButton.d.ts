import { ButtonProps } from '@material-ui/core/Button';

export interface EditButtonProps extends Omit<ButtonProps, 'color' | 'disabled'> {
    /**
    * The title of button.
    */
    title?: string;
    /**
    * Callback fired when a "click" event is detected.
    */
    onClick?: () => void
    /**
    * If true, the button will be disabled.
    */
    disabled?: boolean
    /**
    * @default true
    * If true, the edit icon will be visible.
    */
    editMode?: boolean
    /**
    * @default "theme"
    * The color of button.
    */
    color?: "primary" |
    "info" |
    "theme" |
    "themeNoBackground" |
    "themeWithBackground" |
    "success" |
    "warning" |
    "danger" |
    "rose" |
    "white" |
    "simple" |
    "defaultNoBackground" |
    "primaryNoBackground" |
    "infoNoBackground" |
    "successNoBackground" |
    "warningNoBackground" |
    "dangerNoBackground" |
    "roseNoBackground"
    /**
    * @default "medium"
    * The size of button. 
    */
    size?: "medium" | "small"
    /**
    * @default "medium"
    * The fontSize of button's content.
    */
    fontSize?: "inherit" | "small" | "medium" | "large"
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/edit-button
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/EditButton/EditButton.js
 * 
 * Material-UI component used:
 * 
 * - IconButton: https://material-ui.com/api/icon-button/
 * 
 */
export default function EditButton(props: EditButtonProps): JSX.Element;