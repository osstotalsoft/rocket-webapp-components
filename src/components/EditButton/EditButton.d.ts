import { ButtonProps } from '@material-ui/core/Button';

export interface EditButtonProps extends Omit<ButtonProps, 'color' | 'disabled'> {
    /**
    * The title of button.
    */
    title?: string;
    /**
    * Callback fired when a "click" event is detected.
    */
    onClick?: () => void
    /**
    * If true, the button will be disabled.
    */
    disabled?: boolean
    /**
    * @default true
    * If true, the edit icon will be visible.
    */
    editMode?: boolean
    /**
    * @default "theme"
    * The color of button.
    */
    color?: "primary" |
    "info" |
    "theme" |
    "themeNoBackground" |
    "themeWithBackground" |
    "success" |
    "warning" |
    "danger" |
    "rose" |
    "white" |
    "simple" |
    "defaultNoBackground" |
    "primaryNoBackground" |
    "infoNoBackground" |
    "successNoBackground" |
    "warningNoBackground" |
    "dangerNoBackground" |
    "roseNoBackground" |
    "themeWithGradient"
    /**
    * @default "medium"
    * The size of button. 
    */
    size?: "medium" | "small"
    /**
    * @default "medium"
    * The fontSize of button's content.
    */
    fontSize?: "inherit" | "small" | "medium" | "large"
    /**
    * Override or extend the styles applied to the component
    */
    className?: string
    /**
    * @default false
    * If true, the keyboard focus ripple will be disabled. 
    */
    disableFocusRipple?: boolean
    /**
     * If true, the ripple effect will be disabled.
     */
    disableRipple?: boolean
    /**
     * @default "false"
     * If given, uses a negative margin to counteract the padding on one side 
     * (this is often helpful for aligning the left or right side of the icon 
     * with content above or below, without ruining the border size and shape).
     */
    edge?: "start" | "end" | "false"
}
/**
 *
 * Demos:
 *
 * - https://bit.dev/totalsoft_oss/react-mui/edit-button
 *
 *  Take a look over the code for further information:https://github.com/osstotalsoft/rocket-webapp-components/blob/master/components/EditButton/EditButton.js
 * 
 * Bit component used:
 * 
 * - IconButton: https://bit.dev/totalsoft_oss/react-mui/icon-button
 * 
 */
export default function EditButton(props: EditButtonProps): JSX.Element;