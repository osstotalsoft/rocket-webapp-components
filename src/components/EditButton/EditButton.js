import React from "react";
import PropTypes from "prop-types";
import EditIcon from '@material-ui/icons/Edit';
import VisibilityIcon from '@material-ui/icons/Visibility';
import CustomIconButton from '../IconButton';

const EditButton = ({ title, onClick, editMode, disabled, fontSize, ...rest }) => (
    <CustomIconButton aria-label="Edit" onClick={onClick} disabled={disabled} tooltip={title} {...rest}>
        {editMode === true
            ? <EditIcon fontSize={fontSize} />
            : <VisibilityIcon fontSize={fontSize} />}
    </CustomIconButton>
)

EditButton.defaultProps = {
    editMode: true,
    color: "theme",
    fontSize: "medium",
    size: "medium"
};

EditButton.propTypes = {
    /**
    * The title of button.
    */
    title: PropTypes.string,
    /**
    * Callback fired when a "click" event is detected.
    */
    onClick: PropTypes.func,
    /**
    * If true, the edit icon will be visible.
    */
    editMode: PropTypes.bool,
    /**
    * If true, the button will be disabled.
   */
    disabled: PropTypes.bool,
    /**
    * The color of button.
    */
    color: PropTypes.oneOf([
        "primary",
        "info",
        "theme",
        "themeNoBackground",
        "themeWithBackground",
        "success",
        "warning",
        "danger",
        "rose",
        "white",
        "simple",
        "defaultNoBackground",
        "primaryNoBackground",
        "infoNoBackground",
        "successNoBackground",
        "warningNoBackground",
        "dangerNoBackground",
        "roseNoBackground",
        "themeWithGradient"
    ]),
    /**
    * The size of button.
    */
    size: PropTypes.oneOf([
        "small",
        "medium"
    ]),
    /**
    * The fontSize of button's content.
    */
    fontSize: PropTypes.oneOf([
        "inherit",
        "small",
        "medium",
        "large"
    ])
};

export default EditButton;