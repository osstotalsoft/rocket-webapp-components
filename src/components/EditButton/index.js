import EditButton from "./EditButton";
export default EditButton;