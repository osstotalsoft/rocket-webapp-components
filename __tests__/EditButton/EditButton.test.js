import React from 'react';
import EditButton from '../../components/EditButton/EditButton';
import { mount } from 'enzyme';

describe('EditButton', () => {
    it('the color, fontSize and size props should be equal to the default values if they are not provided', () => {

        const wrapper = mount(<EditButton
            title="Edit"
            onClick={() => { }}
        ></EditButton>);

        expect(wrapper.props().color).toBe('theme');
        expect(wrapper.props().fontSize).toBe('medium');
        expect(wrapper.props().size).toBe('medium');
    });
});