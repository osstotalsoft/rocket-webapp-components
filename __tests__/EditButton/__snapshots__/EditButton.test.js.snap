// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditButton snapshot 1`] = `
<span
  aria-describedby={null}
  className=""
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Edit"
>
  <button
    aria-label="Edit"
    className="MuiButtonBase-root MuiIconButton-root makeStyles-button-1 makeStyles-theme-17"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</span>
`;
